src/pycodehash/dependency/dependency.py

```diff
@@ -0,0 +1,14 @@
+import sys
+from importlib.metadata import version
+
+from pycodehash.datasets.approximate_hasher import ApproximateHasher
+
+
+class DependencyHash(ApproximateHasher):
+    """Get the hash of a list of user-provided dependencies """
+    @staticmethod
+    def collect_metadata(dependencies: list[str]) -> dict[str, str]:
+        for dep in dependencies:
+            if dep not in sys.modules:
+                raise ModuleNotFoundError(f'No module named "{dep}"')
+        return {dep: version(dep) for dep in sorted(dependencies)}
```

Comment:
What does version do when there is no version that can be found? Packages always have versions?

Reply:
I think it's safe. From what I see the version is extracted from the package wheel. And the PEP 491 standard (The Wheel Binary Package Format) requires one. (The version source code assumes the version is present in the metadata.)